x-pack/plugins/actions/server/builtin_action_types/webhook.ts

```diff
@@ -34,10 +34,20 @@ const ConfigSchema = schema.object(configSchemaProps);
 type ActionTypeConfigType = TypeOf<typeof ConfigSchema>;
 
 // secrets definition
-type ActionTypeSecretsType = TypeOf<typeof SecretsSchema>;
-const SecretsSchema = schema.object({
-  user: schema.string(),
-  password: schema.string(),
+export type ActionTypeSecretsType = TypeOf<typeof SecretsSchema>;
+const secretSchemaProps = {
+  user: schema.maybe(schema.string()),
+  password: schema.maybe(schema.string()),
+};
+const SecretsSchema = schema.object(secretSchemaProps, {
+  validate: secrets => {
+    // user and password must be set together (or not at all)
+    if (!secrets.password && !secrets.user) return;
+    if (secrets.password && secrets.user) return;
+    return i18n.translate('xpack.actions.builtin.webhook.invalidUsernamePassword', {
```

Comment:
I'm not sure of config.schema is now generating i18n friendly error messages, but it didn't use to, so it seemed to me to make sense to NOT translate custom validation error messages.

Reply:
I was following what we do for whitelist validation in webhook/slack/pagerduty. I'll test it out to see if it's supported, and if not remove it everywhere.
```
  } catch (whitelistError) {
    return i18n.translate('xpack.actions.builtin.slack.slackConfigurationError', {
      defaultMessage: 'error configuring slack action: {message}',
      values: {
        message: whitelistError.message,
      },
    });
  }
```

Reply:
Translations seem to work ok in config schema. I created a dummy translation for the user/password validation and it shows up:
```
$ curl -s -k -Hkbn-xsrf:true -u elastic:changeme -HContent-Type:application/json -d '{"name":"dummy webhook", "actionTypeId":".webhook", "config":{"url":"http://127.0.0.1:8000"}, "secrets": {"user":"bob"} }' https://localhost:5601/api/action | jq .
{
  "statusCode": 400,
  "error": "Bad Request",
  "message": "error validating action type secrets: (无法处理的密码 Chinese translation goes here)"
}
```